tests/Unit/ApplicationWatchPathsTest.php

```
@@ -0,0 +1,379 @@
+
+        // With the current broken implementation, this returns empty
+        // because it tries to match files starting with literal "!"
+        $matches = $this->matchWatchPathsCurrentBehavior($changed_files, $watch_paths);
+        $this->assertEmpty($matches); // This is why your webhook doesn't trigger!
+
+        // Even if the file had ! in the path, fnmatch would treat ! as a literal character
+        // not as a negation operator, so it still wouldn't match the pattern correctly
+        $changed_files = ['test/file.ts'];
+        $matches = $this->matchWatchPathsCurrentBehavior($changed_files, $watch_paths);
+        $this->assertEmpty($matches);
+    }
```

PHPMD (2.15.0)

The method test_current_broken_behavior_with_negation_patterns is not named in camelCase. (undefined)

(CamelCaseMethodName)